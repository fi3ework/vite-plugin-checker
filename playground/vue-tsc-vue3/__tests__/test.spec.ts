import stringify from 'fast-json-stable-stringify'
import { describe, expect, it } from 'vitest'

import {
  diagnostics,
  editFile,
  expectStderrContains,
  isBuild,
  isServe,
  log,
  resetReceivedLog,
  sleepForEdit,
  sleepForServerReady,
  stripedLog,
} from '../../testUtils'

describe('vue-tsc-vue3', () => {
  describe.runIf(isServe)('serve', () => {
    it('get initial error and subsequent error', async () => {
      await sleepForServerReady(2)
      expect(stringify(diagnostics)).toMatchSnapshot()
      expect(stripedLog).toMatchSnapshot()

      console.log('-- edit file --')
      resetReceivedLog()
      editFile('src/App.vue', (code) =>
        code.replace('<HelloWorld msg1="Diana" />', '<HelloWorld msg2="Diana" />')
      )
      await sleepForEdit(2)
      expect(stringify(diagnostics)).toMatchSnapshot()
      expect(stripedLog).toMatchSnapshot()
    })
  })

  describe.runIf(isBuild)('build', () => {
    it('should fail', async () => {
      const expectedMsg = `src/App.vue(3,4): error TS2322: Type '{ msg1: string; }' is not assignable to type 'ComponentProps<DefineComponent<{ msg: { type: StringConstructor; required: true; }; }, { count: Ref<number>; }, unknown, {}, {}, ComponentOptionsMixin, ComponentOptionsMixin, ... 4 more ..., {}>>'.`
      expectStderrContains(log, expectedMsg)
    })
  })
})
