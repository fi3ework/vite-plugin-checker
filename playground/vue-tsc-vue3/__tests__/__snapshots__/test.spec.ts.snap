// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vue-tsc-vue3 > serve > get initial error and subsequent error 1`] = `"[{"checkerId":"vue-tsc","frame":"    1 | <template>/n    2 |   <img alt=/"Vue logo/" src=/"./assets/logo.png/" />/n  > 3 |   <HelloWorld msg1=/"Diana/" />/n      |    ^^^^^^^^^^/n    4 | </template>/n    5 |/n    6 | <script lang=/"ts/">","id":"<PROJECT_ROOT>/playground-temp/vue-tsc-vue3/src/App.vue","level":1,"loc":{"column":4,"file":"<PROJECT_ROOT>/playground-temp/vue-tsc-vue3/src/App.vue","line":3},"message":"Argument of type '{ msg1: string; }' is not assignable to parameter of type 'Partial<{}> & Omit<{ readonly msg: string; } & VNodeProps & AllowedComponentProps & ComponentCustomProps & Readonly<...>, never> & Record<...>'./n  Property 'msg' is missing in type '{ msg1: string; }' but required in type 'Omit<{ readonly msg: string; } & VNodeProps & AllowedComponentProps & ComponentCustomProps & Readonly<ExtractPropTypes<{ msg: { type: StringConstructor; required: true; }; }>>, never>'.","stack":""}]"`;

exports[`vue-tsc-vue3 > serve > get initial error and subsequent error 2`] = `
[
  "
 ERROR(vue-tsc)  Argument of type '{ msg1: string; }' is not assignable to parameter of type 'Partial<{}> & Omit<{ readonly msg: string; } & VNodeProps & AllowedComponentProps & ComponentCustomProps & Readonly<...>, never> & Record<...>'.
  Property 'msg' is missing in type '{ msg1: string; }' but required in type 'Omit<{ readonly msg: string; } & VNodeProps & AllowedComponentProps & ComponentCustomProps & Readonly<ExtractPropTypes<{ msg: { type: StringConstructor; required: true; }; }>>, never>'.
 FILE  <PROJECT_ROOT>/playground-temp/vue-tsc-vue3/src/App.vue:3:4

    1 | <template>
    2 |   <img alt="Vue logo" src="./assets/logo.png" />
  > 3 |   <HelloWorld msg1="Diana" />
      |    ^^^^^^^^^^
    4 | </template>
    5 |
    6 | <script lang="ts">

[vue-tsc] Found 1 error. Watching for file changes.",
]
`;

exports[`vue-tsc-vue3 > serve > get initial error and subsequent error 3`] = `"[{"checkerId":"vue-tsc","frame":"    1 | <template>/n    2 |   <img alt=/"Vue logo/" src=/"./assets/logo.png/" />/n  > 3 |   <HelloWorld msg1=/"Diana/" />/n      |    ^^^^^^^^^^/n    4 | </template>/n    5 |/n    6 | <script lang=/"ts/">","id":"<PROJECT_ROOT>/playground-temp/vue-tsc-vue3/src/App.vue","level":1,"loc":{"column":4,"file":"<PROJECT_ROOT>/playground-temp/vue-tsc-vue3/src/App.vue","line":3},"message":"Argument of type '{ msg1: string; }' is not assignable to parameter of type 'Partial<{}> & Omit<{ readonly msg: string; } & VNodeProps & AllowedComponentProps & ComponentCustomProps & Readonly<...>, never> & Record<...>'./n  Property 'msg' is missing in type '{ msg1: string; }' but required in type 'Omit<{ readonly msg: string; } & VNodeProps & AllowedComponentProps & ComponentCustomProps & Readonly<ExtractPropTypes<{ msg: { type: StringConstructor; required: true; }; }>>, never>'.","stack":""},{"checkerId":"vue-tsc","frame":"    1 | <template>/n    2 |   <img alt=/"Vue logo/" src=/"./assets/logo.png/" />/n  > 3 |   <HelloWorld msg2=/"Diana/" />/n      |    ^^^^^^^^^^/n    4 | </template>/n    5 |/n    6 | <script lang=/"ts/">","id":"<PROJECT_ROOT>/playground-temp/vue-tsc-vue3/src/App.vue","level":1,"loc":{"column":4,"file":"<PROJECT_ROOT>/playground-temp/vue-tsc-vue3/src/App.vue","line":3},"message":"Argument of type '{ msg2: string; }' is not assignable to parameter of type 'Partial<{}> & Omit<{ readonly msg: string; } & VNodeProps & AllowedComponentProps & ComponentCustomProps & Readonly<...>, never> & Record<...>'./n  Property 'msg' is missing in type '{ msg2: string; }' but required in type 'Omit<{ readonly msg: string; } & VNodeProps & AllowedComponentProps & ComponentCustomProps & Readonly<ExtractPropTypes<{ msg: { type: StringConstructor; required: true; }; }>>, never>'.","stack":""}]"`;

exports[`vue-tsc-vue3 > serve > get initial error and subsequent error 4`] = `
[
  "
 ERROR(vue-tsc)  Argument of type '{ msg2: string; }' is not assignable to parameter of type 'Partial<{}> & Omit<{ readonly msg: string; } & VNodeProps & AllowedComponentProps & ComponentCustomProps & Readonly<...>, never> & Record<...>'.
  Property 'msg' is missing in type '{ msg2: string; }' but required in type 'Omit<{ readonly msg: string; } & VNodeProps & AllowedComponentProps & ComponentCustomProps & Readonly<ExtractPropTypes<{ msg: { type: StringConstructor; required: true; }; }>>, never>'.
 FILE  <PROJECT_ROOT>/playground-temp/vue-tsc-vue3/src/App.vue:3:4

    1 | <template>
    2 |   <img alt="Vue logo" src="./assets/logo.png" />
  > 3 |   <HelloWorld msg2="Diana" />
      |    ^^^^^^^^^^
    4 | </template>
    5 |
    6 | <script lang="ts">

[vue-tsc] Found 1 error. Watching for file changes.",
]
`;
